src/algebra/pointwise.lean

```diff
@@ -169,6 +169,9 @@ lemma empty_mul [has_mul α] : ∅ * s = ∅ := image2_empty_left
 @[simp, to_additive]
 lemma mul_empty [has_mul α] : s * ∅ = ∅ := image2_empty_right
 
+lemma empty_pow [monoid α] (n : ℕ) (hn : n ≠ 0) : (∅ : set α) ^ n = ∅ :=
+by rw [←nat.sub_add_cancel (nat.pos_of_ne_zero hn), pow_succ, empty_mul]
+
```

Comment:
The following lines  should be enough to make `card_pow_eq_card_pow_card_univ` work for most concrete sets `s`:
```suggestion

instance decidable_mem_mul [monoid α] [fintype α] [decidable_eq α]
  [decidable_pred (∈ s)] [decidable_pred (∈ t)] (a : α) :
  decidable (a ∈ s * t) :=
decidable_of_iff _ mem_mul.symm

instance decidable_mem_pow [monoid α] [fintype α] [decidable_eq α]
  [decidable_pred (∈ s)] (n : ℕ) (a : α) :
  decidable (a ∈ (s ^ n)) :=
begin
  induction n with n ih generalizing a,
  { rw [pow_zero, mem_one], apply_instance },
  { letI := ih, rw pow_succ, apply_instance }
end

```

For example:
```lean
example {G : Type*} [group G] [decidable_eq G] [fintype G] (S : set G) (x : G) :
  fintype.card ↥(({x} : set G) ^ (fintype.card G)) = fintype.card ↥(({x} : set G) ^ (fintype.card G)) :=
group.card_pow_eq_card_pow_card_univ ({x} : set G) (fintype.card G) le_rfl
```
works only without the `open_locale classical` below.